actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/BatchOperationNotifierActor.java

```diff
@@ -0,0 +1,335 @@
+      Map<String, Object> requestMap = request.getRequest();
+      batchUpdateOperationNotifier(
+          (CourseBatch) requestMap.get(BackgroundOperations.OLD.name()),
+          (CourseBatch) requestMap.get(BackgroundOperations.NEW.name()));
+    } else if (request
+        .getOperation()
+        .equalsIgnoreCase(ActorOperations.BATCH_OPERATION.getValue())) {
+      Map<String, Object> requestMap = request.getRequest();
+      batchEnrollOperationNotifier(
+          (Map<String, Object>) requestMap.get(BackgroundOperations.COURSE_MAP.name()),
+          (CourseBatch) requestMap.get(BackgroundOperations.COURSE_BATCH.name()),
+          (String) requestMap.get(BackgroundOperations.OPERATION_TYPE.name()));
+    } else {
+      onReceiveUnsupportedOperation(request.getOperation());
+    }
+  }
+
+  /*
+   * This method takes courseBatch and operationType String as parameter and
+   * process invite-only batch enrol and unenroll operation notifications mentor
+   * and learner add remove Operation
+   */
+  private void batchCreateOperationNotifier(CourseBatch courseBatch, String operationType) {
+    if (operationType.equals(JsonKey.ADD)) {
+      sendEmailNotification(courseBatch, JsonKey.ADD);
+    } else if (operationType.equals(JsonKey.REMOVE)) {
+      sendEmailNotification(courseBatch, JsonKey.REMOVE);
+    }
+  }
+
+  /*
+   * This method takes map and courseBatch as parameter and process open batch
+   * enrol and unenroll operation notifications
+   */
+  @SuppressWarnings("unchecked")
+  public void batchEnrollOperationNotifier(
+      Map<String, Object> courseMap, CourseBatch courseBatch, String operationType) {
+    Map<String, Object> requestMap = createRequestMap(courseBatch);
+    List<String> userId = new ArrayList<>();
+    userId.add((String) courseMap.get(JsonKey.USER_ID));
+    List<Map<String, Object>> participentList = getUsersFromDB(userId);
+    Map<String, String> user;
+    if (participentList.size() > 0) {
+      if (operationType.equals(JsonKey.ADD)) {
+        user = getUserData(participentList.get(0), JsonKey.ADD);
+        user.put(JsonKey.TEMPLATE_NAME, JsonKey.BATCH_LEARNER_ENROL);
+      } else {
+        user = getUserData(participentList.get(0), JsonKey.REMOVE);
+        user.put(JsonKey.TEMPLATE_NAME, JsonKey.OPEN_BATCH_LEARNER_UNENROL);
+      }
+      sendMail(user, requestMap);
+    }
+  }
+
+  /*
+   * This method takes courseBatch object and ObjectType as parameter and process
+   * update batch related operation notifications
+   */
+  @SuppressWarnings("unchecked")
+  private void sendEmailNotification(CourseBatch courseBatch, String operationType) {
+    List<Map<String, Object>> mentorList, participentList;
+    mentorList = getUsersFromDB(courseBatch.getMentors());
+    participentList = getUsersFromDB(getParticipants(courseBatch));
+    if (mentorList != null) {
+      for (Map<String, Object> mentor : mentorList) {
+        Map<String, Object> requestMap = new HashMap<String, Object>();
```

Comment:
Almost same code is repeated below , we can create a common method to handle this.

Reply:
done